app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java

```diff
@@ -285,41 +266,72 @@ private void resetAdapter() {
     searchPresenter.getRecentSearches();
   }
 
-  private ArrayAdapter<String> getDefaultAdapter() {
-    return new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1) {
-      @NonNull
-      @Override
-      public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
-        View row;
+  private DefaultAdapter getDefaultAdapter() {
+    return new DefaultAdapter();
+  }
 
-        if (convertView == null) {
-          row = LayoutInflater.from(parent.getContext())
-              .inflate(android.R.layout.simple_list_item_1, null);
-        } else {
-          row = convertView;
-        }
+  public class DefaultAdapter extends RecyclerView.Adapter<DefaultAdapter.ViewHolder> {
```

Comment:
we will want this to be in a separate file and also this probably does not need to exist as it will do the same function as the other adapter which is just render data

Reply:
I didn't got this. What should I do here? Create a new file for this adapter?